### PR TITLE
buildsystem: improve maintainer information handling

### PR DESCRIPTION
**Pull request by feckert**
The two commits are not dependent on each other, but affect the maintainer handling.

- @aparcar remove duplicate MAINTAINER from package-dumpinfo.mk
- @Ansuel @ynezz remove Maintainer info from ipkg control file on the target

### CODE REVIEW

**Comment by feckert** [comment]
@musashino205 thanks done

File: config/Config-build.in [hunk 1]
```diff
@@ -114,6 +114,14 @@ menu "Global build settings"
 		  This makes file checksums part of package metadata. It increases size
 		  but provides you with pkg_check command to check for flash corruptions.
 
+	config IPK_REMOVE_MAINTAINER
+		bool
+		prompt "Remove packages maitainer information on the target"
+		help
+		  Attackers know, that developers have access to source code of the device
```
Comment (aparcar) [inline config/Config-build.in:121]: Don't you think attackers also know, that the source code contain maintainer information? If you try to malicoiously get commit access to source code aka a repository, you likely look at it before (e.g. to understand its impact). In that case one easily obtain that information anyway.

**Review by aparcar — CHANGES_REQUESTED** [review]

File: include/package-dumpinfo.mk [hunk 2]
```diff
@@ -44,7 +44,6 @@ $(if $(KCONFIG),Kernel-Config: $(KCONFIG)
 )$(if $(BUILDONLY),Build-Only: $(BUILDONLY)
 )$(if $(HIDDEN),Hidden: $(HIDDEN)
 )Description: $(if $(Package/$(1)/description),$(Package/$(1)/description),$(TITLE))
-$(MAINTAINER)
```
Comment (aparcar) [inline include/package-dumpinfo.mk:47]: This looks fine to me.

File: config/Config-build.in (hunk 1 above)
Comment (feckert) [reply config/Config-build.in:121]: That's not quite right. This is not about the creator of the source code but about the maintainer of the package at OpenWrt.
I know that you can get the information in other ways, but you don't have to make it too easy?
The creator of the OpenWrt package usually does not know the internals of the upstream package he maintains, but he can add patches, which is an attack vector if you put the OpenWrt package maintainer under pressure.

Comment (aparcar) [reply config/Config-build.in:121]: Sounds a bit like "security by obscurity" to me. Distributions make these information very easy accessible, like [Debian](https://packages.debian.org/sid/devel/tmate), [Alpine](https://pkgs.alpinelinux.org/package/edge/testing/x86_64/tmate) or even [OpenWrt](https://openwrt.org/packages/pkgdata/tmate).

Comment (feckert) [reply config/Config-build.in:121]: Thanks for your feedback. Absolutely right, I agree with you that is "security by obscurity" if that's what you want to call it. The original question from my side was actually do we need the information on the rootfs target?
Isn't it enough that we have it in the repo? Why not always remove it on the rootfs target?

**Comment by feckert** [comment]
Thanks for the feedback.

File: include/rootfs.mk [hunk 3]
```diff
@@ -60,6 +60,12 @@ ifdef CONFIG_CLEAN_IPKG
   endef
 endif
 
+# Remove not needed information from the control file
+# 1: roofs base path
+define clean_control_file
+	-sed -i '/^Maintainer: /d' $(1)/usr/lib/opkg/info/*.control
```
Comment (aparcar) [inline include/rootfs.mk:66]: Please be aware that we eventually plan your switch to APK which renders those changes obsolete. Just a comment

**Review by aparcar — APPROVED** [review]

File: include/rootfs.mk (hunk 3 above)
Comment (feckert) [reply include/rootfs.mk:66]: Thanks for the hint.
If this is the case then I will discard the newly opened PR this commit, that is removing this Information from the control file. What information is stored on the rootfs target when using APK? Is there any information on this yet?

**Pull request closed without merging**